ocs_ci/ocs/monitoring.py

```
@@ -53,49 +58,66 @@ def validate_pvc_created_and_bound_on_monitoring_pods():
-    """
-    Validate monitoring pods are respinned and running state
-
-    Args:
-        pod_list (list): List of the pods where pvc are mounted
-    """
-    ocp = OCP(api_version='v1', kind='Pod', namespace='openshift-monitoring')
-    assert ocp.wait_for_resource(
-        condition=constants.STATUS_PENDING, resource_name=pods_list[0]
-    ), f"failed to reach pod {pods_list[0]} "
-    f"desired status {constants.STATUS_PENDING}"
-    for pod in pods_list:
-        assert ocp.wait_for_resource(
-            condition=constants.STATUS_RUNNING, resource_name=pod
-        ), f"failed to reach pod {pod} "
-        f"desired status {constants.STATUS_RUNNING}"
-
-
 def get_list_pvc_objs_created_on_monitoring_pods():
     """
     Returns list of pvc objects
+
     """
-    pvc_list = get_all_pvcs(namespace='openshift-monitoring')
+    pvc_list = get_all_pvcs(namespace=defaults.OCS_MONITORING_NAMESPACE)
     ocp_pvc_obj = OCP(
-        kind=constants.PVC, namespace='openshift-monitoring'
+        kind=constants.PVC, namespace=defaults.OCS_MONITORING_NAMESPACE
     )
     pvc_obj_list = []
     for pvc in pvc_list['items']:
         pvc_dict = ocp_pvc_obj.get(resource_name=pvc.get('metadata').get('name'))
         pvc_obj = PVC(**pvc_dict)
         pvc_obj_list.append(pvc_obj)
     return pvc_obj_list
+
+
+def get_kube_pod_spec_volumes_persistentvolumeclaims_info_metric():
+    """
+    Returns the created pvc information
+
+    """
+    prometheus = ocs_ci.utility.prometheus.PrometheusAPI()
+    response = prometheus.get(
+        'query?query=kube_pod_spec_volumes_persistentvolumeclaims_info'
+    )
+    return json.loads(response.content.decode('utf-8'))
+
+
+@retry(UnexpectedBehaviour, tries=10, delay=3, backoff=1)
+def collected_metrics_for_created_pvc(pvc_name):
+    """
+    Checks whether initially pvc related data is collected on pod
+
+    Args:
+        pvc_name (str): Name of the pvc
+
```

add return/raise values

---

done

---

also raise UnexpectedBehaviour

---

@Akarsha-rai could you please add the raise UnexpectedBehaviour ?